### PR TITLE
Auth task

### PR DESCRIPTION
**Pull request by yanzacd**
## Summary

Se adiciona la gema devise auth para la autenticacion de los usuarios

## Trello Card

https://trello.com/c/dVWvDezV/24-implementar-autenticación

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
   devise_for :users
+  # token auth routes available at /api/v1/auth
+  namespace :api do
```
Comment (mtejedorwolox) [inline config/routes.rb]: use versionist `api_version`

File: Gemfile [hunk 2]
```diff
@@ -63,3 +63,5 @@ end
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem 'devise'
+gem 'devise_token_auth'
+gem 'omniauth', '~> 1.8', '>= 1.8.1'
```
Comment (mtejedorwolox) [inline Gemfile]: why?

**Review by mtejedorwolox — CHANGES_REQUESTED** [review]
you can't delete migrations once they have run. The correct way to do it is to create a new migration (without deleting the previous ones) and make there the changes you need for it to work properly.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class User < ApplicationRecord
+  # Include default devise modules.
+  devise  :database_authenticatable, :registerable, :recoverable, :rememberable,
+          :trackable, :validatable, :confirmable, :omniauthable
```
Comment (mtejedorwolox) [inline app/models/user.rb]: remove confirmable and omniauthable

**Review by mtejedorwolox — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 1 above)
Comment (yanzacd) [reply config/routes.rb]: Para usar el api_version seria de esta manera: (No estoy seguro)

namespace :api do
   mount_devise_token_auth_for 'User', at: 'auth'
      api_version(
        header: {
          name: 'Accept',
          value: 'application/vnd.versionist_api.v1+json'
        },
        module: 'V1',
        default: true,
        defaults: { format: :json }
      ) do
        resources :books, only: [:index, :show]
      end
    end
end

File: Gemfile (hunk 2 above)
Comment (yanzacd) [reply Gemfile]: Solo quito la gema de omniauth? o también la de devise_toke_auth ?

**Review by mtejedorwolox — APPROVED** [review]

**Pull request merged**